packages/teleport-plugin-vue-base-component/src/utils.ts

```diff
@@ -329,8 +333,16 @@ const createVuePropsDefinition = (uidlPropDefinitions: Record<string, UIDLPropDe
         )
     }
 
-    acc[name] =
-      typeof defaultValue !== 'undefined' ? { type: mappedType, default: defaultValue } : mappedType
+    let defaultPropValue = null
+
+    if (defaultValue !== 'undefined') {
+      const defaultValueAsFunc = new ParsedASTNode(
```

Comment:
as a small optimization thing, you could only compute defaultValueAsFunc when the type is `object` or `array`